### ISSUE TITLE
Update CodeOwners for Surface components

### ISSUE THREAD

**Issue by tclune** [GEOS-ESM/GEOSgcm_GridComp#266]
@mathomp4  I forwarded the spreadsheet of assignments separately.
(cc'ing @gmao-rreichle )

**Comment by mathomp4** [comment]
@gmao-rreichle I think I could maybe do this, but CODEOWNERS is really more designed for ownership of *directories* not files, or at most globs of files (`*.py`, or the like). 

Per-file CODEOWNERS means that eventually the file will go stale and the default owner will start owning all new files.

**Comment by mathomp4** [comment]
Also: I'd need to know what to call your new Teams.

**Comment by gmao-rreichle** [comment]
@mathomp4 @tclune : I knew something wasn't going to be as straightforward as I would have liked...
In that case, I think we should proceed as described below, although I'm not sure about the "Shared" directory.  See question below.

There should just be a single team of land CodeOwners.
The new team should be called "Land Team" or, more precisely, "Land Science Team", unless that's too long.
The new team should initially consist of @rdkoster , @gmao-jkolassa and @gmao-rreichle .
The new land gatekeeper should be added later, once she's completed onboarding.  
This "Land [Science] Team" should "own" the following directories and all sub-directories within these directories:
- `GEOSsurface_GridComp/GEOSland_GridComp`
- `GEOSsurface_GridComp/GEOSlake_GridComp`  
- `GEOSsurface_GridComp/Shared`

I understand that this gives us co-ownership of CMake files.  We'll be happy to defer to the CMake Team on these.

Re: `GEOSsurface_GridComp/Shared`  -- this directory is shared with the landice and ocean groups.  It includes files of no interest to us (`OASIMalbedoMod.f`) and files that are truly shared across groups (`GEOS_SurfaceGridComp.rc`, `StieglitzSnow.F90`, `SurfParams.F90`).

I'm not sure how best to assign CodeOwners in this case.  I suppose if the "Land [Science] Team" owns the entire directory, other teams (e.g., a "Landice Team") could co-own the entire directory, and we'll sort out approval on a case-by-case basis.

**Comment by tclune** [comment]
1.  I think LandTeam is sufficient.    If we ever need to make fine distinction about land, we can always revisit.
2. I defer to Matt as to whether the 3 files in Shared are sufficiently unique to justify single file entries in the CodeOwners file.    Failing that, it really is not so bad to "own" the entire directory.   Your team would just glance at PR's that modify other files in that directory and see that your stuff is not touched and happily click "Approve".    I suppose it depends a bit on how often the other files are modified.

**Comment by mathomp4** [comment]
@gmao-rreichle Actually, I believe how I would set it up that @GEOS-ESM/cmake-team would still own the `CMakeLists.txt` files. The `CODEOWNERS` file is "ordered" in that the last match wins. We usually keep the CMake ownership last so it always falls to us.

First, I've created @GEOS-ESM/land-team which is the "Jana" team for lack of a better descriptor. On your spreadsheet, you have two teams (the "Jana" and "Sarith" teams). What is a good name for the second one? "Land Restart Team"? I can give that team ownership of the mk_restarts and Raster directories. It would be nice to have Sarith notified of changes there (though I suppose Sarith is the one that would make those changes 😄 )

**Comment by mathomp4** [comment]
Also, we don't have a landice group (yet?) in the org, so for now, I can't distinguish those. But we do have an Ocean Team, so I can give them rights to that one file in Shared.

**Comment by gmao-rreichle** [comment]
@mathomp4 , @tclune : When @mathomp4 said that we should think of CodeOwners more in terms of directories, I wasn't sure whether we can assign different CodeOwners to sub-directories within the Surface GridComp.  It sounds like this can be done.  
I don't think there should be individuals as CodeOwners.  That's likely to become problematic if someone happens to be on leave.
Given this, it makes sense to create a 2nd team including Randy, Sarith, and myself, plus the new land gatekeeper at a later time, and perhaps representatives from the ocean and landice groups.  
I suggest calling this team the "Surface-Preproc-Team".  
The "Surface-Preproc-Team" should own the following directories:

- `GEOSsurface_GridComp/GEOSland_GridComp/GEOScatch_GridComp/mk_restarts`
- `GEOSsurface_GridComp/Shared/Raster`

(Note that `GEOSroute_GridComp` stays with "Land-Team", and the four files in `GEOSsurface_GridComp/Shared/` should also stay with individual discipline teams as discussed above.)

Note also that I'm not sure why `mk_restarts/`, which is for _all_ of Surface, resides in `GEOScatch_GridComp/` -- see #232.   Wouldn't `mk_restarts/` be better placed in  `GEOSsurface_GridComp/Shared/`, next to `Raster/`?  It would be helpful if someone from the SI team could take a look at addressing #232.

Finally, perhaps check with Richard Cullather and Bin about their interest in being CodeOwners for  `GEOSlandice_GridComp/`

**Comment by mathomp4** [comment]
@gmao-rreichle Okay. I noticed the Route GC is labeled for the Sarith group (aka @GEOS-ESM/surface-preproc-team ) on the spreadsheet, but I'll keep it in @GEOS-ESM/land-team per your instructions here.

**Comment by smahanam** [comment]
I am not sure I understand everything that is going on. I did not know  whether I was even a code owner - Randy was the lead scientist, I have been maintaining land science developments.

If this ownership change means transferring the ongoing work to the @GEOS-ESM/land-team, both GEOScatchCN_GridComp and GEOSroute_GridComp are not ready.

GEOScatchCN_GridComp: I created and maintained but lost the track during CVS-git transfer so I need to verify few things and make sure we transferred from CVS to git correctly. And there is another pending issue, I can prepare this transfer soon.

GEOSroute sure needs me to finish it. We have a ticket with the SI-team, once that issue is resolved only I can finish GEOSroute_GridComp
